### ISSUE TITLE
Priority algorithm isn't starting with weight algorithm output

### ISSUE THREAD

**Issue by Opeyem1a** [Teamable-Analytics/algorithms#224]
Even when the MAX_ITERATE is set to 0, priority ranks drastically lower than weight on the priority satisfaction metric.
I suspect this is because something about the logic of generating its initial team set is broken

**Comment by Opeyem1a** [comment]
after super deep debugging, I think the weight algorithm is capable of adding duplicate students to a team

**Comment by Opeyem1a** [comment]
tldr the issue is more that the weight algorithm's score was getting inflated artificially. This is because we were persisting the algorithm class instance for each call to `AlgorithmRunner.generate()`, and algorithm classes store an internal `self.teams`

So what was happening is that the algorithms would just append extra students to existing teams (i.e. there are more students in the team set than students in the class, and teams in the team set have duplicate ids). How this happens is not the concern because it's really all due to the fact that a new algorithm class instance should be created every time we want a team set
